File: src/main/kotlin/com/geonyeong/insurance/entity/KindCompanyEntity.java
package com.geonyeong.insurance.entity;


import com.fasterxml.jackson.annotation.JsonBackReference;
import lombok.Data;
import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;
import java.util.List;

@Entity
@Table(name = "kind_company")
@Getter
@Setter
public class KindCompanyEntity {

    @Id
    private @Column(name = "id") Long id;

    public KindCompanyEntity() {
    }

    public KindCompanyEntity(Long id) {
        this.id = id;
    }

    @ManyToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "kind_id")
    private KindEntity kindEntity;

    @ManyToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "company_id")
    private CompanyEntity companyEntity;

    @OneToMany(mappedBy = "kindCompanyEntity",cascade = CascadeType.ALL)
    @JsonBackReference
    private List<GuideEntity> guideEntities;

}

File: src/main/kotlin/com/geonyeong/insurance/entity/KindCompanyEntity.kt
package com.geonyeong.insurance.entity

import com.fasterxml.jackson.annotation.JsonBackReference
import javax.persistence.*

@Entity
@Table(name = "kind_company")
data class KindCompanyEntity(
        @Id
        @Column(name = "id")
        var id: Long,

        @ManyToOne(cascade = arrayOf(CascadeType.ALL))
        @JoinColumn(name = "kind_id")
        var kindEntity: KindEntity?,

        @ManyToOne(cascade = arrayOf(CascadeType.ALL))
        @JoinColumn(name = "company_id")
        var companyEntity: CompanyEntity?,

        @OneToMany(mappedBy = "kindCompanyEntity",cascade = arrayOf(CascadeType.ALL))
        @JsonBackReference
        var guideEntities: List<GuideEntity>?
){
        constructor():this(0,null,null,null)
}